src/components/navbar/navDropdown/NavDropdown.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Link } from 'react-router-dom';
+import classes from './NavDropdown.module.scss';
+import PropTypes from 'prop-types';
+import NavLinkSection from '../navLinkSection/NavLinkSection';
+
+const NavDropdown = ({
+    display,
+    navLink,
+    logo,
+    closeDropdownNav,
+    toggleNewsModal,
+}) => {
+    const initTransitionStyle = {
+        transform: 'translate3d(0, -110%, 0)',
+        visibility: 'hidden',
+    };
+    const terminalTransitionStyle = {
+        transform: 'translate3d(0, 0, 0)',
+        visibility: 'visible',
+    };
+    const [transitionStyle, setTransitionStyle] = useState(initTransitionStyle);
+    const ref = useRef();
+
+    useEffect(() => {
+        // mounting animation
+        if (display) {
+            setTransitionStyle((prevState) => {
+                return {
+                    ...prevState,
+                    ...terminalTransitionStyle,
+                };
+            });
+        } else {
+            setTransitionStyle((prevState) => {
+                return {
+                    ...prevState,
+                    ...initTransitionStyle,
+                };
+            });
+        }
+
+        // click outside to close component
+        function handleClickOutside(event) {
+            //if click is navBurgerContainer (ref.current.parentElement.children[2]) or if click is ref target (.dropdownContainer)
+            if (event.target === ref.current.parentElement.children[2]) return;
+            else if (ref.current && !ref.current.contains(event.target)) {
```

Comment:
If you're returning in an `if` statement it's a bit cleaner to not use `else if`.

```
if (event.target === ref.current.parentElement.children[2]) {
  return;
}

if (ref.current && !ref.current.contains(event.target)) {
  closeDropdownNav();
}
```